client/src/main/scala/org/apache/celeborn/client/LifecycleManager.scala

```diff
@@ -386,7 +415,18 @@ class LifecycleManager(val appUniqueId: String, val conf: CelebornConf) extends
                   // otherwise will lost original exception message
                   context.reply(response)
                 }
-              case PartitionType.REDUCE => context.reply(response)
+              case PartitionType.REDUCE =>
+                val rpcContext: RpcCallContext = context.context
+                if (rpcContext.isInstanceOf[LocalNettyRpcCallContext]) {
+                  context.reply(response)
+                } else {
+                  val serializedMsg =
```

Comment:
the `serializedMsg` should be put to the start of the method to avoid serialization for each request